### ISSUE TITLE
Call an user-provided action on didInsertElement/willDestroyElement

### ISSUE THREAD

**Issue by cibernox** [yapplabs/ember-wormhole#49]
I have a component with a template like this:
```hbs
{{#if isOpen}}
  {{#ember-wormhole to=to renderInPlace=renderInPlace}}
    <div class="{{my-class}}">{{yield}}</div>
  {{/ember-wormhole}}
{{/if}}
```

When the component gets opened/closed, I need to add some events to the root of the page.
I was thinking that it would be very handy to be able to instruct `ember-wormhole` to notify me when it gets added/removed to the dom.

My idea was `{{#ember-wormhole to=to renderInPlace=renderInPlace onInsert=(action "foo") onDestroy=(action "bar")}}`.

The changes to the component on the code would just be

```js
didInsertElement() {
  // ... usual code
  this.sendAction('onInsert');
},

willDestroyElement() {
  // ... usual code
  this.sendAction('onDestroy');
}
```

Would you consider such addition?

**Comment by lukemelia** [comment]
@cibernox I'm not closed to the idea, but it seems a little awkward. You could achieve something similar by including your own component within the wormhole block which has actions on didInsertElement/willDestroyElement. Would that be more or less semantically appropriate in your use case?

**Comment by cibernox** [comment]
It is true that I could create a third component that is rendered inside the wormhole like this:

```hbs
{{#if isOpen}}
  {{#ember-wormhole to=to renderInPlace=renderInPlace}}
    {{third-component}}
  {{/ember-wormhole}}
{{/if}}
```

However, that component would only exist with the sole purpose of letting me know when the wormhole has been rendered or removed from DOM, which is what I suggested just having two actions that the component calls when that happens.

Something I could do now is use `didReceiveAttrs` to check when the component goes from opened to closed and schedule some work in the `afterRender` queue, it feels hacky and would make fastboot integration less straightforward.

**Comment by lukemelia** [comment]
@cibernox Upon reflection, I am amenable to adding these actions, but... I think that they most often would not be the appropriate choice for end developers. In your simplified example, IMO whatever action resulted in isOpen changing is the action that should be used to do what you wanted to with adding/removing events to the root of the page.

**Comment by cibernox** [comment]
@lukemelia Yes, I also reached the conclusion that I should probably use 
`scheduleOnce('afterRender', this, this.reposition)` inside the `open` action. I think that this could work, but I haven't tried yet. 

There is a few more problems to consider. By example, removing events on the `willDestroyElement` of the parent component, since the entire component can be removed without any `close` action being called.

Either way, I've found that even if I really need this approach, it's pretty trivial to implement this myself: https://github.com/cibernox/ember-basic-dropdown/blob/master/addon/components/basic-dropdown/wormhole.js

Given how easy is to implement yourself and that this might lead the users towards bad practices (is very easy cause a double render if the user uses `this.set` on the action called from the `didInsertElement`) I retire this proposal.
